client/src/unifycr.c

```diff
@@ -2080,47 +2080,19 @@ int unifycr_mount(const char prefix[], int rank, size_t size,
 }
 
 /**
- * unmount the mounted file system, triggered
- * by the root process of an application
- * ToDo: add the support for more operations
- * beyond terminating the servers. E.g.
- * data flush for persistence.
+ * unmount the mounted file system
+ * TODO: Add support for unmounting more than
+ * one filesystem.
  * @return success/error code
  */
 int unifycr_unmount(void)
 {
-#if 0
-    int cmd = COMM_UNMOUNT;
-    int bytes_read = 0;
-    int rc;
-    int *response = NULL;
-
-    memset(cmd_buf, 0, sizeof(cmd_buf));
-    memcpy(cmd_buf, &cmd, sizeof(int));
-
-    rc = __real_write(cmd_fd.fd, cmd_buf, sizeof(cmd_buf));
-    if (rc <= 0)
-        return UNIFYCR_FAILURE;
-
-    cmd_fd.events = POLLIN | POLLPRI;
-    cmd_fd.revents = 0;
-
-    rc = poll(&cmd_fd, 1, -1);
-    if (rc < 0)
-        return UNIFYCR_FAILURE;
-
-    if (cmd_fd.revents != 0 && cmd_fd.revents == POLLIN) {
-        bytes_read = __real_read(cmd_fd.fd, cmd_buf, sizeof(cmd_buf));
-
-        response = (int *) cmd_buf;
-
-        if (bytes_read <= 0 ||
-            response[0] != COMM_UNMOUNT || response[1] != ACK_SUCCESS)
-            return UNIFYCR_FAILURE;
-    }
-#endif
+    int ret;
 
-    return UNIFYCR_SUCCESS;
+    /* invoke unmount rpc */
+    printf("calling unmount\n");
+    ret = unifycr_client_unmount_rpc_invoke(&unifycr_rpc_context);
+    return ret;
 }
```

Comment:
Just occurred to me that we should have the client also tear down its end of the connection to the server here, detach from its request and receive shared memory regions and close its socket.

Reply:
@dsikich based on your email, it possible that this logic was never in the client.  In that case, it will take some effort to add it.  We could probably finalize this PR and create an issue to track the fact that we should clean up the connection on the client side.